src/psnr.jl

```diff
@@ -0,0 +1,67 @@
+"""
+    PeakSignalNoiseRatio <: FullReferenceIQI
+    psnr(x, ref [, peakval])
+    assess(PSNR(), x, ref, [, peakval])
```

Comment:
Three sets of api are provided: 

* short name: `psnr(x, ref)`, this is used by other languages most
* struct as function: `PSNR()(x, ref)`, the real definition goes here
* through method `assess`, this is just a wrapper of the second.

Reply:
The idea here is to "**assess** image quality of **x** using method **PSNR** with reference **ref**", as a protocol of any Image quality assessment methods. 

This also follows the idea in https://github.com/JuliaImages/Images.jl/issues/767#issuecomment-438866145

An easy approach to implement the third api is passing arguments into `iqi(x, ref)` (second api). This avoids hacking into dispatches. <--> In `Distances`, they directly write implementations in `evaluate`, and some additional efforts are needed to fix the ambiguities

https://github.com/JuliaImages/ImageQualityIndexes.jl/blob/b2420c3d8555a6ece6d0f7ccf413c224c1931a86/src/generic.jl#L50-L57

Actually, these three lines of codes are identical since `ref` can be absorbed into `args...`, but it satisfies the need though. In the future, I'm planning to move this `generic.jl` file to `ImagesAPI` so that different quality indexes packages can collaborate with the same `assess` api.

---

A more ambitious vision on `Images.jl` in a generic sense:

Since most image processing algorithms and functions are pure-function, and many of them are filters, we can adopt the "network" concept from deep learning. With the second API, we can do something like

```julia
denoise_flow = Flow(AdditiveGaussianNoise(0.0, 0.1),
    MotionBlur(0.1),
    L2TVDenoise(),
    [PSNR(), SSIM()])
denoise_flow(img) # returns psnr and ssim
denoise_flow(img; accumulate=true) # returns all intermediate results
```
Visualization of the workflow can be a promising as well.

Reply:
> Will there be some FullReferenceIQI that require a parameter?

Yes there is. The original definition of structural similarity is `SSIM(x, y) = [l(x,y)]^{α} * [c(x,y)]^{β} * [s(x,y)]^{γ}`, where `α=β=γ=1` and `C₃=C₂/2` are used as default params.

Reference: 

Wang, Z., Bovik, A. C., Sheikh, H. R., & Simoncelli, E. P. (2004). _Image quality assessment: from error visibility to structural similarity_. IEEE transactions on image processing, 13(4), 600-612.